ta4j-core/src/main/java/org/ta4j/core/analysis/criteria/pnl/AverageLossCriterion.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2014-2017 Marc de Verdelhan, 2017-2021 Ta4j Organization & respective
+ * authors (see AUTHORS)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package org.ta4j.core.analysis.criteria.pnl;
+
+import org.ta4j.core.BarSeries;
+import org.ta4j.core.Position;
+import org.ta4j.core.TradingRecord;
+import org.ta4j.core.analysis.criteria.AbstractAnalysisCriterion;
+import org.ta4j.core.analysis.criteria.NumberOfLosingPositionsCriterion;
+import org.ta4j.core.num.Num;
+
+/**
+ * Average gross loss (with commissions).
+ */
+public class AverageLossCriterion extends AbstractAnalysisCriterion {
+
+    private NumberOfLosingPositionsCriterion numberOfLosingPositionsCriterion = new NumberOfLosingPositionsCriterion();
+    private GrossLossCriterion grossLossCriterion = new GrossLossCriterion();
+
+    @Override
+    public Num calculate(BarSeries series, Position position) {
+        Num numberOfLosingTrades = numberOfLosingPositionsCriterion.calculate(series, position);
+        if (numberOfLosingTrades.isZero()) {
+            return series.numOf(0);
+        }
+        Num grossLoss = grossLossCriterion.calculate(series, position);
+        if (grossLoss.isZero()) {
+            return series.numOf(0);
+        }
+        return grossLoss.dividedBy(numberOfLosingTrades);
+    }
+
+    @Override
+    public Num calculate(BarSeries series, TradingRecord tradingRecord) {
+        Num numberOfLosingTrades = numberOfLosingPositionsCriterion.calculate(series, tradingRecord);
+        if (numberOfLosingTrades.isZero()) {
+            return series.numOf(0);
+        }
+        Num grossLoss = grossLossCriterion.calculate(series, tradingRecord);
```

Comment:
This will require 2 iterations over the trading record items. Even though, the streams do have a small performance impact, this version is probably slower, but we can run a micro benchmark... I am actually intrigued, what the difference would be...

Reply:
@petrmac Good idea. You can create a new PR (with your improvements, if any) after this PR is merged. I will not change the whole implementation of this PR. The "2 iterations over the trading record items" can be improved by you after this is merged.